src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -506,8 +506,25 @@ int32_t get_default_ddr_index(int32_t dev_index, int32_t cu_index) {
 }
 
 void xma_enable_mode1(void) {
+    if (xrt_core::config::get_xma_mode2()) {
+        xma_logmsg(XMA_ERROR_LOG, XMAUTILS_MOD, "xma_enable_mode1: Mode2 is selected. Use either mode1 or mode2");
```

Comment:
Error message is a little confusing...could you elaborate a little bit more.  Something like:

> xma_enable_mode1: Mismatch between modes - The config currently is using mode2, but the code flow is enabling mode1.

Reply:
These are debug modes. Either mode1 or mode2 can be used for an application run. Made change to the code. Safety check if mode is already changed.